frame/support/src/traits.rs

```diff
@@ -1551,6 +1554,43 @@ pub trait OnRuntimeUpgrade {
 		0
 	}
 
+	/// Generate a storage key unique to this runtime upgrade.
+	///
+	/// This can be used to communicate data from pre-upgrade to post-upgrade state and check
+	/// them. See [`set_temp_storage`] and [`get_temp_storage`].
+	#[cfg(feature = "try-runtime")]
+	fn storage_key(ident: &str) -> [u8; 32] {
```

Comment:
Why do we need this methods as part of `OnRuntimeUpgrade`?

Reply:
THe use case is: You want to write some data into an arbitrary storage in `pre-upgrade`, then read it back in `post-migrate` and do some checking. This is just a helper for that use case.

Reply:
Could we please implement this extension trait? By that I mean
```
#[cfg(feature = "try-runtime")]
pub trait SomeName {
 // you functions
}

impl<T: OnRuntimeUpgrade> SomeName for T {}
```

Or could this not just be free standing functions? I don't see why they need to be part of the trait?